Add export scene codelens & wizard

Splines opened the pull request:
Closes #38.

### Feature

Here we add the functionality to export a scene as described [in my comment here](https://github.com/Manim-Notebook/manim-notebook/issues/38#issuecomment-2453216114). Or rather to be guided through some configuration to then obtain an executable command. This command will be pasted into a new terminal such that the user can review it and then execute it by pressing enter. This also allows to add any further flags before executing the command.

![image](https://github.com/user-attachments/assets/6ac95c1c-8c4d-4bc7-90d8-60b04bae6628)


### Known limitations
The action is triggered by a new codelens `Export Scene` at any line of Python code that defines a class. To limit this to classes that define Manim scenes, we can tackle #12 with the heuristic [described there](https://github.com/Manim-Notebook/manim-notebook/issues/12#issuecomment-2442317826) in a future PR.

Splines commented on src/utils/multiStepQuickPickUtil.ts:
```diff
@@ -0,0 +1,219 @@
+
+////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * What follows is helper code that eases multi-step input flow in quick picks
+ * by wrapping the VSCode API, e.g. it implements a stack to allow for back
+ * navigation in the quick picks.
+ * 
+ * This code is copied 1:1 from the VSCode extension samples [1]. We do not use
+ * every feature of it, however it's designed flexibly enough that we can
+ * pick and choose what we need.
+ * 
+ * We could reduce the code size by removing unused features, but we leave it as
+ * is for now to keep the code as close to the original example as possible.
+ * If necessary, we might want to use more features of it in the future.
+ * See it as kind of a library that should (IMHO) be included in the VSCode API
+ * itself.
+ * 
+ * 🛑 List of what was manually changed after copying the code:
+ * - Export the `MultiStepInput` class.
+ * 
+ * [1] https://github.com/microsoft/vscode-extension-samples/blob/6446fd7012bd1f7ef107683e09e488e943e668ef/quickinput-sample/src/multiStepInput.ts#L131C1-L314C2
+ */
```
Note this comment

bhoov commented:
It works! 🎉🎉🎉

My initial impressions after playing around with it:
1. As a keyboard-oriented user, I kept looking for a command in the palette called "Manim Notebook: Export current scene" before realizing this functionality was only a button in the UI. I like the design philosophy that searching the command palette for "Manim Notebook" returns all the primary interactions with this tool. Does it make sense to include this as a cmd?
2. The workflow would benefit from some indication to the user that we are just "generating the command for export" instead of doing the actual export -- you and I have the advantage of knowing what the expected workflow is. Few new users will. Some suggestions:

1. Rename the "Export Scene" button (and palette command) to "Generate cmd to export scene"
2. Show some alert to user indicating "Export command pasted in terminal"

Additionally, if we stick with this workflow, perhaps we should also copy the export command to the clipboard so users are able to play around with additional settings without losing the generated command. Also useful if a user does not want to use the built in terminal (e.g., a very long scene will take a while to generate and they want it to run in a dedicated terminal in a background)

Splines commented:
Thanks for your review 🙏, yes both points absolutely make sense. I've addressed them in the newest commits. Please also review my "Note this comment" thread regarding the multi-step quick pick util file and resolve if d'accord.

Splines commented:
By the way, you can also post such comments as a review with the option `Request changes` in the `Files changed` tab. This way, I can re-request a review from you once I've addressed all the points (which is now the case).

bhoov commented:
> you can also post such comments as a review with the option Request changes in the Files changed tab

Sounds good @Splines! I have not yet figured the optimal way to exchange feedback on Github, thanks for your tip! :)

I'm playing around with the changes you kindly implemented per my feedback, and I'm starting to prefer your original, simpler "Export" wording that you had before... Especially with the new alerts that are added, I think the process is more self-explanatory. Please see my small updates and see if you like the phrasing.

1. Include "Manim scene" to parallel "Manim cell" elsewhere
2. Specify "current scene" to palette command
